File: Java/com.faculdadevincit.designpatterns.commandpattern/src/ControleRemoto.java
public class ControleRemoto {
    private ICommand comand;

    public ICommand getComand() {
        return comand;
    }

    public void setComand(ICommand comand) {
        this.comand = comand;
    }

    public void pressionarBotao() {
        comand.executar();
    }

    public void desfazerPressionar() {
        comand.desfazer();
    }
}

File: CSharp/FaculdadeVincit.DesignPatterns.CommandPattern/ControleRemoto.cs
namespace FaculdadeVincit.DesignPatterns.CommandPattern;

public class ControleRemoto
{
    public ICommand? Command { get; set; }
    public void PressionarBotao() => Command?.Executar();
    public void DesfazerPressionar() => Command?.Desfazer();
}
